src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/SymbolUsageInfo.cs

```diff
@@ -13,14 +14,19 @@ namespace Microsoft.CodeAnalysis
     /// For namespaces and types, this corresponds to values from <see cref="TypeOrNamespaceUsageInfo"/>.
     /// For methods, fields, properties, events, locals and parameters, this corresponds to values from <see cref="ValueUsageInfo"/>.
     /// </summary>
+    [DataContract]
     internal readonly struct SymbolUsageInfo : IEquatable<SymbolUsageInfo>
     {
         public static readonly SymbolUsageInfo None = Create(ValueUsageInfo.None);
 
+        [DataMember(Order = 0)]
         public ValueUsageInfo? ValueUsageInfoOpt { get; }
+
+        [DataMember(Order = 1)]
         public TypeOrNamespaceUsageInfo? TypeOrNamespaceUsageInfoOpt { get; }
 
-        private SymbolUsageInfo(ValueUsageInfo? valueUsageInfoOpt, TypeOrNamespaceUsageInfo? typeOrNamespaceUsageInfoOpt)
+        // Must be public since it's used for deserialization.
```

Comment:
This comment is unnecessary. It should be linked to https://github.com/dotnet/runtime/issues/42412 or removed.